python/pyspark/ml/param/__init__.py

```diff
@@ -439,12 +441,26 @@ def _set(self, **kwargs):
             self._paramMap[p] = value
         return self
 
+    def _clearParam(self, param):
+        """
+        Clears a param from the param map if it has been explicitly set.
+        """
+        if self.isSet(param):
+            del self._paramMap[param]
+
     def _setDefault(self, **kwargs):
         """
         Sets default params.
         """
         for param, value in kwargs.items():
-            self._defaultParamMap[getattr(self, param)] = value
+            p = getattr(self, param)
+            if value is not None and not isinstance(value, JavaObject):
```

Comment:
True, but I hope we can get rid of that after the stopwords issue is handled.  Conversions from Java should not be handled by setDefault.